lib/dialects/abstract/query-interface.js

```diff
@@ -480,7 +481,7 @@ class QueryInterface {
       context: 'changeColumn',
       table: tableName,
     });
-    const sql = this.queryGenerator.changeColumnQuery(tableName, query);
```

Comment:
`dataTypeOrOptions` should not be passed if it's a DataType instead of `ModelAttributeColumnOptions`. Checking if it's a plain object (lodash has a isPlainObject function) should be enough